yang/iana-ipv6-ext-types.yang

```diff
@@ -0,0 +1,154 @@
+    "Internet Assigned Numbers Authority
+
+     ICANN
+     12025 Waterfront Drive, Suite 300
+     Los Angeles, CA 90094
+     
+
+     Tel: +1 424 254 5300
+
+     <mailto:iana@iana.org>";
+
+  description
+    "This YANG module translates IANA registry 'IPv6 Extension Header
+     Types' to YANG derived types.
+
+     Copyright (c) 2023 IETF Trust and the persons identified as
+     authors of the code. All rights reserved.
+
+     Redistribution and use in source and binary forms, with or
+     without modification, is permitted pursuant to, and subject to
+     the license terms contained in, the Revised BSD License set
+     forth in Section 4.c of the IETF Trust's Legal Provisions
+     Relating to IETF Documents
+     (https://trustee.ietf.org/license-info).
+
+     This version of this YANG module was generated from the
+     corresponding IANA registry using an XSLT stylesheet from the
+     'iana-yang' project (https://github.com/llhotka/iana-yang).";
+
+  reference
+    "Internet Protocol Version 6 (IPv6) Parameters
+     (https://www.iana.org/assignments/ipv6-parameters/)";
+
+  revision 2023-09-29 {
+    description
+      "Current revision as of the revision date specified in the XML
+       representation of the registry page.";
+    reference
+      "https://www.iana.org/assignments/ipv6-parameters/ipv6-parameters.xml";
+  }
+
+  /* Typedefs */
+
+  typedef ipv6-extension-header-type-name {
+    type enumeration {
+      enum IPv6Hop-by-HopOption {
+        value 0;
+        description
+          "IPv6 Hop-by-Hop Option";
+        reference
+          "RFC 8200";
+      }
+      enum RoutingHeaderforIPv6 {
+        value 43;
+        description
+          "Routing Header for IPv6";
+        reference
+          "- RFC 8200
+           - RFC 5095";
+      }
+      enum FragmentHeaderforIPv6 {
+        value 44;
+        description
+          "Fragment Header for IPv6";
+        reference
+          "RFC 8200";
+      }
+      enum EncapsulatingSecurityPayload {
+        value 50;
+        description
+          "Encapsulating Security Payload";
+        reference
+          "RFC 4303";
+      }
+      enum AuthenticationHeader {
+        value 51;
+        description
+          "Authentication Header";
+        reference
+          "RFC 4302";
+      }
+      enum DestinationOptionsforIPv6 {
+        value 60;
+        description
+          "Destination Options for IPv6";
+        reference
+          "RFC 8200";
+      }
+      enum MobilityHeader {
+        value 135;
+        description
+          "Mobility Header";
+        reference
+          "RFC 6275";
+      }
+      enum HostIdentityProtocol {
+        value 139;
+        description
+          "Host Identity Protocol";
+        reference
+          "RFC 7401";
+      }
+      enum Shim6Protocol {
+        value 140;
+        description
+          "Shim6 Protocol";
+        reference
+          "RFC 5533";
+      }
+      enum Useforexperimentationandtesting {
```

Comment:
I would add a filter to remove 253/254 as these are not supposed to be leaked outside.